### PR TITLE
Revise 21-0966 Confirmation page logic

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary

This corrects some sloppy logic that I had previously committed for the Confirmation page of 21-0966. It is much more rigorously tested and I've done pretty extensive spot checks locally in the browser. The logic in here is more complex than I had originally thought it would be. This code is partially-refactored so if anyone wants to grind away on it to tighten it up, feel free. But I've already dumped a lot of time into some code which I *hope* will not have to change much. I'm not proud. But the tests should provide a good safety net if we want to refactor the implementation code.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/703

## Testing done

lots. tons. but it's never truly enough is it? 🤔 

## What areas of the site does it impact?

Confirmation page of 21-0966

### CODE REVIEW

File: src/applications/simple-forms/21-0966/config/helpers.js [hunk 1]
```diff
@@ -115,6 +116,87 @@
...
+export const getInfoAlertText = (data, alreadySubmittedIntents) => {
+  const benefitSelections = Object.keys(data.benefitSelection).filter(
+    key => data.benefitSelection[key],
+  );
+  let benefitSelection = benefitSelections[0].toLowerCase();
+  const expirationDate = new Date(
+    alreadySubmittedIntents[benefitSelections[0].toLowerCase()].expirationDate,
+  ).toDateString();
+  if (benefitSelections.length > 1) {
+    benefitSelection = 'compensationAndPension';
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/21-0966/config/helpers.js:211]: I think this is a red herring.  This var IS getting used.  I'm guessing CodeQL didn't think so just 'cuz it's being referenced down there as an array-key.

File: src/applications/simple-forms/21-0966/config/helpers.js [hunk 2]
```diff
@@ -115,115 +115,194 @@ export const initializeFormDataWithPreparerIdentification = preparerIdentificati
...
-  }
-  if (
-    benefitSelection.includes(veteranBenefits.compensation) &&
-    benefitSelection.includes(veteranBenefits.pension)
-  ) {
-    return 'disability compensation and pension claims';
+const isAlreadySubmitted = (data, alreadySubmittedIntents) => {
+  const newlySelectedBenefit = benefitSelections(data).filter(
+    benefitSelection => {
+      if (alreadySubmittedBenefitIntents(alreadySubmittedIntents).length > 0) {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/21-0966/config/helpers.js:129]: hm i don't have a specific recommendation here, but i did find this a bit hard to read. no change necessary. I think the more return statements you have in general, the more confusing a function becomes.

**Review by rhasselle-oddball — COMMENTED** [review]
don't see any issues , seems good to me

File: src/applications/simple-forms/21-0966/config/helpers.js (hunk 2 above)
Comment (Thrillberg) [reply src/applications/simple-forms/21-0966/config/helpers.js]: Thanks @rhasselle-oddball ! Does it look better now?

**Review by tlei123 — APPROVED** [review]
LGTM

**Pull request merged**